### PR TITLE
feat: Upgrade JHipster, Quarkus dependencies, fix broken flows

### LINKED ISSUE #213: Migrating Mongo to DevServices
**Describe the bug**

Today the generated application that use MongoDB as a primary database, use Mongock for testing. We should use DevServices instead. But some migration is needed

**Migrating Configuration**

Today the Mongo configuration is as follow:

```
jhi.mongodb.port=27017
%test.jhi.mongodb.port=37017
jhi.mongodb.host=localhost
%prod.jhi.mongodb.host=localhost
quarkus.mongodb.connection-string=mongodb://${jhi.mongodb.host}:${jhi.mongodb.port}
quarkus.mongodb.database=dummy
```

The end result should look like:

```
%test.quarkus.mongodb.devservices.image-name=mongo:4.2.7
%test.quarkus.mongodb.devservices.port=27017
quarkus.mongodb.database=dummy
%prod.quarkus.mongodb.connection-string=mongodb://localhost:27017
```

**Migrating Test Configuration**

`MongoDbTestResource`.

**Pom.xml**

We can get all the dependencies on MongoDB and Mongock and only leave `quarkus-mongodb-panache`

### LINKED ISSUE #222: JHipster Quarkus can't build native image with latest version of GraalVM (21.2.0.r11-grl)
If I try to perform the steps in [this demo script](https://github.com/oktadev/okta-jhipster-quarkus-example/blob/main/demo.adoc) (which is based on [this blog post](https://developer.okta.com/blog/2021/03/08/jhipster-quarkus-oidc)) with GraalVM 21.2.0.r11-grl. It seems to work, but running the executable does not work.

```
sdk install java 21.2.0.r11-grl
```

If I downgrade to `21.0.0.2.r11-grl`, it works:

```
sdk default java 21.0.0.2.r11-grl
```

**To Reproduce**

```
sdk install java 21.2.0.r11-grl
gu install native-image
git clone https://github.com/oktadev/okta-jhipster-quarkus-example.git
cd okta-jhipster-quarkus-example
./mvnw package -Pnative -DskipTests
./target/*runner # this will fail
```

There is a stack trace when building, but it seems like it succeeds:

```
[INFO] --- quarkus-maven-plugin:1.12.1.Final:build (default) @ rocket ---
[INFO] [org.jboss.threads] JBoss Threads version 3.2.0.Final
[INFO] [org.hibernate.Version] HHH000412: Hibernate ORM core version 5.4.28.Final
[INFO] [io.quarkus.deployment.pkg.steps.JarResultBuildStep] Building native image source jar: /Users/mraible/Downloads/jhipster-quarkus/target/rocket-1.0.0-SNAPSHOT-native-image-source-jar/rocket-1.0.0-SNAPSHOT-runner.jar
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Building native image from /Users/mraible/Downloads/jhipster-quarkus/target/rocket-1.0.0-SNAPSHOT-native-image-source-jar/rocket-1.0.0-SNAPSHOT-runner.jar
[ERROR] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Unable to get GraalVM version from the native-image binary.
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] -J-DCoordinatorEnvironmentBean.transactionStatusManagerEnable=false -J-Dsun.nio.ch.maxUpdateArraySize=100 -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dcom.sun.xml.bind.v2.bytecode.ClassTailor.noOptimize=true -J-Dvertx.logger-delegate-factory-class-name=io.quarkus.vertx.core.runtime.VertxLogDelegateFactory -J-Dvertx.disableDnsResolver=true -J-Dio.netty.leakDetection.level=DISABLED -J-Dio.netty.allocator.maxOrder=1 -J-Duser.language=en -J-Duser.country=US -J-Dfile.encoding=UTF-8 --initialize-at-run-time=com.mycompany.myapp.security.RandomUtil --initialize-at-build-time= -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy\$BySpaceAndTime -H:+JNI -H:+AllowFoldMethods -jar rocket-1.0.0-SNAPSHOT-runner.jar -H:FallbackThreshold=0 -H:+ReportExceptionStackTraces -H:-AddAllCharsets -H:EnableURLProtocols=http,https --enable-all-security-services --no-server -H:-UseServiceLoaderFeature -H:+StackTrace rocket-1.0.0-SNAPSHOT-runner
Warning: Ignoring server-mode native-image argument --no-server.
[rocket-1.0.0-SNAPSHOT-runner:16117]    classlist:   4,546.27 ms,  1.69 GB
--initialize-at-build-time without arguments has been deprecated and will be removed in GraalVM 22.0.
[rocket-1.0.0-SNAPSHOT-runner:16117]        (cap):   2,648.53 ms,  1.69 GB
[rocket-1.0.0-SNAPSHOT-runner:16117]        setup:   4,529.07 ms,  1.69 GB
java.lang.NoClassDefFoundError: com/oracle/svm/core/jdk/LocalizationFeature
	at io.quarkus.runner.AutoFeature.beforeAnalysis(AutoFeature.zig:1525)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$11(NativeImageGenerator.java:691)
	at com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:71)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:691)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:532)
	at com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:491)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:380)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:543)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:119)
	at com.oracle.svm.hosted.NativeImageGeneratorRunner$JDK9Plus.main(NativeImageGeneratorRunner.java:573)
Caused by: java.lang.ClassNotFoundException: com.oracle.svm.core.jdk.LocalizationFeature
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 10 more
13:17:33,132 INFO  [org.hib.val.int.uti.Version] HV000001: Hibernate Validator 6.2.0.Final
13:17:41,195 INFO  [org.hib.Version] HHH000412: Hibernate ORM core version 5.4.28.Final
[rocket-1.0.0-SNAPSHOT-runner:16117]     (clinit):     439.50 ms,  4.47 GB
[rocket-1.0.0-SNAPSHOT-runner:16117]   (typeflow):   9,570.27 ms,  4.47 GB
[rocket-1.0.0-SNAPSHOT-runner:16117]    (objects):   8,955.04 ms,  4.47 GB
[rocket-1.0.0-SNAPSHOT-runner:16117]   (features):     744.96 ms,  4.47 GB
[rocket-1.0.0-SNAPSHOT-runner:16117]     analysis:  20,587.39 ms,  4.47 GB
[rocket-1.0.0-SNAPSHOT-runner:16117]     universe:   1,389.63 ms,  4.47 GB
[rocket-1.0.0-SNAPSHOT-runner:16117]      (parse):   1,058.93 ms,  4.47 GB
[rocket-1.0.0-SNAPSHOT-runner:16117]     (inline):   2,709.61 ms,  5.75 GB
[rocket-1.0.0-SNAPSHOT-runner:16117]    (compile):  11,848.93 ms,  5.79 GB
[rocket-1.0.0-SNAPSHOT-runner:16117]      compile:  17,162.05 ms,  5.79 GB
[rocket-1.0.0-SNAPSHOT-runner:16117]        image:   7,385.27 ms,  5.79 GB
[rocket-1.0.0-SNAPSHOT-runner:16117]        write:     962.30 ms,  5.79 GB
[rocket-1.0.0-SNAPSHOT-runner:16117]      [total]:  57,299.11 ms,  5.79 GB
# Printing build artifacts to: /Users/mraible/Downloads/jhipster-quarkus/target/rocket-1.0.0-SNAPSHOT-native-image-source-jar/rocket-1.0.0-SNAPSHOT-runner.build_artifacts.txt
[WARNING] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] objcopy executable not found in PATH. Debug symbols will not be separated from executable.
[WARNING] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] That will result in a larger native image with debug symbols embedded in it.
[INFO] [io.quarkus.deployment.QuarkusAugmentor] Quarkus augmentation completed in 63466ms
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:18 min
[INFO] Finished at: 2021-08-10T13:18:20-06:00
[INFO] ------------------------------------------------------------------------
Execution time: 1 min. 21 s.
```

**Expected behavior**

I expect that the native image would work, just like it does with 21.0.0.2.r11-grl. 

**Desktop (please complete the following information):**

-   OS: Mac
-   Version Big Sur v11.4

### LINKED ISSUE #228: Support latest version of Quarkus
**Is your feature request related to a problem? Please describe.**
Yes, I need to run JHipster by the latest version of Quarkus which has lots of features.

**Describe the solution you'd like**
I worked a bit on this template of the project and it seems, there are a few things to be able to use Quarkus 2.* e.g.
Eclipse ConfigProperty and ConfigMapping

**Describe alternatives you've considered**
As a JHispter developer, I would like to use the latest version of Quarkus instead of version 1.* of this framework.

**Additional context**
there is not any additional.

### LINKED ISSUE #245: Upgrade to JHipster 7.9.3
**Is your feature request related to a problem? Please describe.**

Upgrade to the latest release of JHipster. This might require refactoring to use the latest blueprint architecture.

### PR DESCRIPTION
**Pull request by vishal423**
Closes #245 
Closes #213 
Closes #228 
Closes #222

### CODE REVIEW

**Comment by vishal423** [comment]
@mraible, there seems an issue with embedded mongodb dependency and driver version. This is independent of current upgrade and can be tracked separately ( I believe there exists a ticket for mongodb migration).

**Comment by vishal423** [comment]
@mraible, did you get a chance to review the changes? I don't have commit access to this repository to assign reviewers.

I am amidst another change to bump quarkus ecosystem dependencies and noticed a few more broken flows. I will raise separate PRs to fix those.

**Comment by avdev4j** [comment]
@vishal423 it seems the way GitHub allows us to manage groups is a bit complicated: as administrator I can create a team but I can't add people to existing ones if I'm not the maintainer of this team...

That doesn't make sense... So I've added you directly with the writing access. Let's see after how to fix that.

**Comment by mraible** [comment]
@vishal423 I'm traveling and attending conferences this week. I'll take a look when I'm back home next week.

**Comment by vishal423** [comment]
thanks @avdev4j  for granting write access

@mraible, please take your time, no rush

File: .editorconfig [hunk 1]
```diff
@@ -13,5 +13,4 @@ insert_final_newline = true
 trim_trailing_whitespace = false
 
 [{package,bower}.json]
-indent_style = space
```
Comment (mraible) [inline .editorconfig:16]: Does this mean that indents are now tabs, or indents are spaces by default?

**Comment by mraible** [comment]
I tried this branch today with @danielpetisme's [tutorial from the Okta developer blog](https://developer.okta.com/blog/2021/03/08/jhipster-quarkus-oidc). 

The initial console output is a bit lengthy, especially since there are two logos displayed. Also, there are a lot of "Using blueprint" messages.

Screen Shot 2022-10-12 at 13 00 36

At the end of the application creating part, there's a printout of commands:

```
> jhipster@0.0.1-SNAPSHOT clean-www
> rimraf target/classes/META-INF/resourcesapp/{src,target/}
```

It looks like there should be a space between `resources` and `app`, but not sure. Also, it seems weird that files are removed just after installation.

I was using Java 17 and received an error:

```
[INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (enforce-versions) @ jhipster ---
[WARNING] Rule 1: org.apache.maven.plugins.enforcer.RequireJavaVersion failed with message:
You are running an incompatible version of Java. JHipster Quarkus supports JDK
                                    8 to
                                    13.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

Reverting to Java 11 fixes this. However, I think we should support Java 17 since the main generator does.

Finally, after starting Keycloak and running `./mvnw`, I get an error:

```
2022-10-12 13:05:38,606 WARN  [io.qua.dat.dep.dev.DevServicesDatasourceProcessor] (build-41) Unable to start devservices for default datasource as this datasource type (h2) does not support devservices
2022-10-12 13:05:38,680 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.agroal.deployment.AgroalProcessor#build threw an exception: io.quarkus.runtime.configuration.ConfigurationException: Unable to find a JDBC driver corresponding to the database kind 'h2' for the default datasource. Either provide a suitable JDBC driver extension, define the driver manually, or disable the JDBC datasource by adding 'quarkus.datasource.jdbc=false' to your configuration if you don't need it.
	at io.quarkus.agroal.deployment.AgroalProcessor.resolveDriver(AgroalProcessor.java:332)
	at io.quarkus.agroal.deployment.AgroalProcessor.getAggregatedConfigBuildItems(AgroalProcessor.java:278)
	at io.quarkus.agroal.deployment.AgroalProcessor.build(AgroalProcessor.java:83)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:820)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2442)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1476)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:415)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:275)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:66)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:91)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:424)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:64)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:133)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:90)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:145)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:63)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.agroal.deployment.AgroalProcessor#build threw an exception: io.quarkus.runtime.configuration.ConfigurationException: Unable to find a JDBC driver corresponding to the database kind 'h2' for the default datasource. Either provide a suitable JDBC driver extension, define the driver manually, or disable the JDBC datasource by adding 'quarkus.datasource.jdbc=false' to your configuration if you don't need it.
	at io.quarkus.agroal.deployment.AgroalProcessor.resolveDriver(AgroalProcessor.java:332)
	at io.quarkus.agroal.deployment.AgroalProcessor.getAggregatedConfigBuildItems(AgroalProcessor.java:278)
	at io.quarkus.agroal.deployment.AgroalProcessor.build(AgroalProcessor.java:83)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:820)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2442)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1476)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.builder.Execution.run(Execution.java:116)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:151)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:413)
	... 9 more
Caused by: io.quarkus.runtime.configuration.ConfigurationException: Unable to find a JDBC driver corresponding to the database kind 'h2' for the default datasource. Either provide a suitable JDBC driver extension, define the driver manually, or disable the JDBC datasource by adding 'quarkus.datasource.jdbc=false' to your configuration if you don't need it.
	at io.quarkus.agroal.deployment.AgroalProcessor.resolveDriver(AgroalProcessor.java:332)
	at io.quarkus.agroal.deployment.AgroalProcessor.getAggregatedConfigBuildItems(AgroalProcessor.java:278)
	at io.quarkus.agroal.deployment.AgroalProcessor.build(AgroalProcessor.java:83)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:820)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2442)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1476)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

Below is my `.yo-rc.json`:

```
{
  "generator-jhipster": {
    "applicationType": "monolith",
    "authenticationType": "oauth2",
    "baseName": "jhipster",
    "blueprints": [
      {
        "name": "generator-jhipster-quarkus",
        "version": "2.0.0-beta.1"
      }
    ],
    "buildTool": "maven",
    "cacheProvider": "caffeine",
    "clientFramework": "angularX",
    "clientPackageManager": "npm",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "creationTimestamp": 1665600995874,
    "cypressAudit": true,
    "cypressCoverage": true,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "devServerPort": 4200,
    "dtoSuffix": "DTO",
    "enableGradleEnterprise": false,
    "enableHibernateCache": true,
    "enableTranslation": true,
    "entitySuffix": "",
    "jhiPrefix": "jhi",
    "jhipsterVersion": "7.9.3",
    "languages": ["en", "nl"],
    "nativeLanguage": "en",
    "otherModules": [
      {
        "name": "generator-jhipster-quarkus",
        "version": "2.0.0-beta.1"
      }
    ],
    "packageName": "com.company.myapp",
    "pages": [],
    "prodDatabaseType": "mysql",
    "reactive": false,
    "searchEngine": false,
    "serverPort": "8080",
    "serviceDiscoveryType": false,
    "skipCheckLengthOfIdentifier": false,
    "skipClient": false,
    "skipFakeData": false,
    "skipUserManagement": true,
    "testFrameworks": ["cypress"],
    "websocket": false,
    "withAdminUi": true
  }
}
```

**Comment by avdev4j** [comment]
Hi @mraible
Quickly, the console print are coming the blueprint itself, so I think we can edit it and remove the console print from each index.js files.
Then, the Java version is fixed in the pom.xml file, so it has been decided in time where we did not choose to use Java 17 in the main generator. Before doing that we might have to check if it works with quarkus libraries (no ideas, even if I think it should be).

**Comment by vishal423** [comment]
>The initial console output is a bit lengthy, especially since there are two logos displayed. Also, there are a lot of "Using blueprint" messages.

The first logo is coming from the JHipster generator itself and I don't see an option to customize/hide that. If you refer to any other blueprint using the latest published JHipster version, it should be the same behavior. The `Using blueprint` logs / warnings are also from the JHipster and can't do much over here.

>It looks like there should be a space between resources and app, but not sure. Also, it seems weird that files are removed just after installation.

It also doesn't look correct to me. Either space or slash should come. I will take a look into that. However, files removal is the JHipster generator behavior and isn't controlled over here 😞 

> Reverting to Java 11 fixes this. However, I think we should support Java 17 since the main generator does.

This PR doesn't touch on the Quarkus lib upgrade. I have a separate unpublished branch that contains the fix to support Java 17, and Quarkus lib upgrades for Maven, however, the OIDC flow seems to be broken. I didn't spend much time debugging the issue, however, that seems to come due to quarkus integration with the problem library. Unfortunately, I am a bit occupied and wouldn't get much time before mid-November. If you want to combine those changes, I can push them over here.

File: .editorconfig (hunk 1 above)
Comment (vishal423) [reply .editorconfig:16]: It was redundant as we define to use spaces for all type of files:
```
[*]
indent_style = space
```

**Comment by mraible** [comment]
> This PR doesn't touch on the Quarkus lib upgrade. 

OK. I'll take a look and see if I can figure out the problem with MongoDB.

> I have a separate unpublished branch that contains the fix to support Java 17, and Quarkus lib upgrades for Maven, however, the OIDC flow seems to be broken. I didn't spend much time debugging the issue, however, that seems to come due to quarkus integration with the problem library. Unfortunately, I am a bit occupied and wouldn't get much time before mid-November. If you want to combine those changes, I can push them over here.

Yes, please push your changes into another PR. If it's easier, you can merge this one first.

File: generators/app/index.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ const chalk = require('chalk');
 const AppGenerator = require('generator-jhipster/generators/app');
 const constants = require('generator-jhipster/generators/generator-constants');
 const packagejs = require('../../package.json');
+const { askForTestOpts } = require('./prompts');
 
 module.exports = class extends AppGenerator {
```
[the remark this answers is not in the document]
Comment (vishal423) [reply generators/app/index.js:8]: Can you point to a specific code function/statement of concern? Overall, it overrides to prints the `Quarkus` logo and customizes the Test options to allow Cypress configuration while disabling Gatling/Cucumber support.

Other priorities are default. Do you want to remove default priorities to cleanup this file code?

**Comment by vishal423** [comment]
I will push those lib changes over here, so that you can take a consolidated view of change. I haven't looked into mongo db flow and the dependencies upgrade.

Additionally, I have replaced the in-memory Redis with devservices support (uses TestContainer), so, docker is going to be a requirement.

File: generators/app/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mraible) [reply generators/app/index.js:8]: I was just pointing out the general structure of the files, not anything specific.

**Comment by mraible** [comment]
I created https://github.com/vishal423/generator-jhipster-quarkus/pull/1. It seems to fix MongoDB, but still has issues.

File: generators/app/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (vishal423) [reply generators/app/index.js:8]: ok. for now I just leave it as is. There is much more work to make this blueprint functional with all supported options :)

**Comment by vishal423** [comment]
I have merged the `MongoDB` fixes, however, my previous library upgrade seems to have broke tests. I will spend some time to see what caused that behavior.

**Comment by vishal423** [comment]
We are now good on tests front except `MongoDB`

**Comment by mraible** [comment]
@vishal423 I was able to fix the MongoDB compilation error. However, tests still fail so it seems whatever authentication mechanism the tests use is not working.

**Comment by mshima** [comment]
> The first logo is coming from the JHipster generator itself and I don't see an option to customize/hide that. If you refer to any other blueprint using the latest published JHipster version, it should be the same behavior. The `Using blueprint` logs / warnings are also from the JHipster and can't do much over here.

@vishal423 you can generate a new cli using `jhipster generate-blueprint` and override the `printLogo`:
https://github.com/jhipster/generator-jhipster/blob/9d929768523d7e5a1a263e6640c19d5b8018b1d2/cli/program.cjs#L257

at:
https://github.com/jhipster/generator-jhipster/blob/2267fe5edf354fc6566f28bc0f7a26e639e5bd46/generators/generate-blueprint/templates/cli/cli.mjs.ejs#L24-L27

**Comment by mshima** [comment]
> > It looks like there should be a space between resources and app, but not sure. Also, it seems weird that files are removed just after installation.
> 
> It also doesn't look correct to me. Either space or slash should come. I will take a look into that. However, files removal is the JHipster generator behavior and isn't controlled over here 😞

This was a pretty old behavior, dropped for v8.

**Comment by vishal423** [comment]
> @vishal423 you can generate a new cli using `jhipster generate-blueprint` and override the `printLogo`: https://github.com/jhipster/generator-jhipster/blob/9d929768523d7e5a1a263e6640c19d5b8018b1d2/cli/program.cjs#L257
> 
> at: https://github.com/jhipster/generator-jhipster/blob/2267fe5edf354fc6566f28bc0f7a26e639e5bd46/generators/generate-blueprint/templates/cli/cli.mjs.ejs#L24-L27

sure @mshima I will take a look. Can we also suppress `Using blueprint` logs? Those are useful during debug, however, doesn't present good user experience on application generation.

**Comment by vishal423** [comment]
> @vishal423 I was able to fix the MongoDB compilation error. However, tests still fail so it seems whatever authentication mechanism the tests use is not working.

Last time I checked, tests were trying to connect on the wrong port than configured. I will take the latest update to see if that's still the issue after your change.

**Comment by mshima** [comment]
> > @vishal423 you can generate a new cli using `jhipster generate-blueprint` and override the `printLogo`: https://github.com/jhipster/generator-jhipster/blob/9d929768523d7e5a1a263e6640c19d5b8018b1d2/cli/program.cjs#L257
> > at: https://github.com/jhipster/generator-jhipster/blob/2267fe5edf354fc6566f28bc0f7a26e639e5bd46/generators/generate-blueprint/templates/cli/cli.mjs.ejs#L24-L27
> 
> sure @mshima I will take a look. Can we also suppress `Using blueprint` logs? Those are useful during debug, however, doesn't present good user experience on application generation.

Sure

**Comment by mraible** [comment]
> I will take the latest update to see if that's still the issue after your change.

I tried upgrading to Mongock 5 and [ran into issues](https://github.com/mongock/mongock/issues/582).

**Comment by vishal423** [comment]
@mraible, If you are fine, I will revert the latest MongoDB changes made on this PR to use the latest stable released `Mongock` version (`4.x`) available on the maven repository. I was able to use DevServices (in place of in-memory as requested in this ticket https://github.com/jhipster/generator-jhipster-quarkus/issues/213) and fix test issues observed for `AccountResource`. Same fix can be leveraged to fix other entities as well.

**Comment by mraible** [comment]
@vishal423 I'm OK with this.

**Comment by vishal423** [comment]
@mraible, this PR is now ready to be reviewed. During testing, I noticed a few broken scenarios and raised a couple of issues, however much more can be identified if regressed further. I will raise couple more issues that I noticed recently.

**Comment by mraible** [comment]
I should have time to test in the next 24 hours. I'm currently at JavaOne, so things are a bit hectic.On Oct 17, 2022, at 13:50, Vishal Mahajan ***@***.***> wrote:﻿
@mraible, this PR is now ready to be reviewed. During testing, I noticed a few broken scenarios and raised a couple of issues, however much more can be identified if regressed further. I will raise couple more issues that I noticed recently.

**Comment by mraible** [comment]
@vishal423 I tried again with the `.yo-rc.json` in https://github.com/jhipster/generator-jhipster-quarkus/pull/247#issuecomment-1276628070. Everything launches OK with `./mvnw`:

```
2022-10-19 11:19:37,061 WARN  [io.qua.hib.orm.dep.HibernateOrmProcessor] (build-24) Hibernate ORM is disabled because no JPA entities were found
2022-10-19 11:19:37,071 WARN  [io.net.res.dns.DnsServerAddressStreamProviders] (build-89) Can not find io.netty.resolver.dns.macos.MacOSDnsServerAddressStreamProvider in the classpath, fallback to system defaults. This may result in incorrect DNS resolutions on MacOS.
2022-10-19 11:19:37,111 INFO  [io.qua.oid.dep.dev.OidcDevConsoleProcessor] (build-89) OIDC Dev Console: discovering the provider metadata at http://localhost:9080/realms/jhipster/.well-known/openid-configuration

        ██╗ ██╗   ██╗ ████████╗ ███████╗   ██████╗ ████████╗ ████████╗ ███████╗
        ██║ ██║   ██║ ╚══██╔══╝ ██╔═══██╗ ██╔════╝ ╚══██╔══╝ ██╔═════╝ ██╔═══██╗
        ██║ ████████║    ██║    ███████╔╝ ╚█████╗     ██║    ██████╗   ███████╔╝
  ██╗   ██║ ██╔═══██║    ██║    ██╔════╝   ╚═══██╗    ██║    ██╔═══╝   ██╔══██║
  ╚██████╔╝ ██║   ██║ ████████╗ ██║       ██████╔╝    ██║    ████████╗ ██║  ╚██╗
   ╚═════╝  ╚═╝   ╚═╝ ╚═══════╝ ╚═╝       ╚═════╝     ╚═╝    ╚═══════╝ ╚═╝   ╚═╝

:: JHipster 🤓  :: https://www.jhipster.tech ::

                                                 Powered by Quarkus 2.13.0.Final
2022-10-19 11:19:38,181 INFO  [io.agr.pool] (Quarkus Main Thread) Datasource '<default>': Initial size smaller than min. Connections will be created when necessary

2022-10-19 11:19:38,318 INFO  [liq.database] (Quarkus Main Thread) Set default schema name to PUBLIC
2022-10-19 11:19:38,454 INFO  [liq.changelog] (Quarkus Main Thread) Reading from PUBLIC.DATABASECHANGELOG
2022-10-19 11:19:38,482 INFO  [liq.lockservice] (Quarkus Main Thread) Successfully acquired change log lock
2022-10-19 11:19:38,510 INFO  [liq.lockservice] (Quarkus Main Thread) Successfully released change log lock
2022-10-19 11:19:38,571 INFO  [io.quarkus] (Quarkus Main Thread) jhipster 1.0.0-SNAPSHOT on JVM (powered by Quarkus 2.13.0.Final) started in 1.998s. Listening on: http://localhost:8080
2022-10-19 11:19:38,572 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2022-10-19 11:19:38,572 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [agroal, cache, cdi, hibernate-orm, hibernate-orm-panache, hibernate-validator, jdbc-h2, liquibase, mailer, micrometer, narayana-jta, oidc, qute, resteasy, resteasy-jsonb, resteasy-problem, security, servlet, smallrye-context-propagation, smallrye-health, smallrye-openapi, swagger-ui, vertx]
```

If I go to `http://localhost:8080` and click sign in, I get a 401 in my browser.

```json
{
  "status": 401,
  "title": "Unauthorized",
  "instance": "/oauth2/authorization/oidc"
}
```

**Comment by vishal423** [comment]
#248 is to track this issue. One workaround is to remove the problem library and associated code references (that I used to test OIDC flow).

**Review by mraible — APPROVED** [review]

**Comment by vishal423** [comment]
I will merge by tomorrow. @mraible, I assume you got chance to verify all issues closed as part of it.

**Comment by vishal423** [comment]
@pascalgrimaud , bounty claimed https://opencollective.com/generator-jhipster/expenses/110561

**Comment by DanielFran** [comment]
@vishal423 approved

**Pull request merged**